frontend/src/components/widgets/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,255 @@
+/**
+ * @license
+ * Copyright 2018-2020 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import { shallow } from "enzyme"
+import { fromJS } from "immutable"
+import { sliderOverrides } from "lib/widgetTheme"
+import { Slider as UISlider } from "baseui/slider"
+import { WidgetStateManager } from "lib/WidgetStateManager"
+
+import Slider, { Props } from "./Slider"
+
+jest.mock("lib/WidgetStateManager")
+
+const sendBackMsg = jest.fn()
+const getProps = (elementProps: object = {}): Props => ({
+  element: fromJS({
+    id: 1,
+    label: "Label",
+    format: "%d",
+    default: [5],
+    min: 0,
+    max: 10,
+    step: 1,
+    ...elementProps,
+  }),
+  width: 0,
+  disabled: false,
+  widgetMgr: new WidgetStateManager(sendBackMsg),
+})
+
+function timeout(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+describe("Slider widget", () => {
+  it("should show a label", () => {
+    const props = getProps()
+    const wrapper = shallow(<Slider {...props} />)
+
+    expect(wrapper.find("label").text()).toBe("Label")
+  })
+
+  it("should send the value to the backend when did mount", async () => {
+    const props = getProps()
+
+    const wrapper = shallow(<Slider {...props} />)
+
+    await timeout(200)
```

Comment:
Why is this needed? Maybe add comment.

Reply:
The slider component sets the widget value using a debounce of 200ms, that's why this is needed